tests/utils_tests/test_archive.py

```diff
@@ -32,8 +32,6 @@ def test_extract_function(self):
     def test_extract_file_permissions(self):
         """archive.extract() preserves file permissions."""
         mask = stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO
-        umask = os.umask(0)
-        os.umask(umask)  # Restore the original umask.
```

Comment:
I'm not sure what you expected here, but this was added to fetch the umask on the system the tests are being run on so that it could be applied below to avoid failures where systems have different umask configured.

Reply:
I understand was the intention was, however it was only being applied for the test on the `no_permissions` file, which probably should have a **minimum** of permissions so the umask does not apply. 
Unless I am really misunderstanding what the intended behavior is of course, could you maybe share your thoughts on that?

Having said that, looking at it again I see the umask *does* need to be applied when testing directory creation, where it currently isn't - agree?
(For this to work, `_copy_permissions` also needs to be applied to extracted directories)

Again, appreciate the comments, sorry for having switched gears too fast being eager to start contributing :)

Reply:
@pope1ni : 
I gave this a bit more thought: 
Given archive.Archive is only used for providing templates (startapp, startproject) - in those scenarios my guess is we do not really care about preserving original permissions in the archive beyond making sure the extracted files are readable/executable (to avoid issues such as ticket #26494 and ticket #27628. Do you agree?

If so, my suggestion would be to:
1) Update _copy_permissions to use bit-wise OR to ensure read rights 
2) Update the test to check if extracted files have read rights
3) Update the test to check if extracted *dir* has `0o777 & ~umask`

Would appreciate your input! :)